scipy/spatial/transform/rotation.py

```diff
@@ -1461,6 +1464,84 @@ def magnitude(self):
         else:
             return angles
 
+    def reduce(self, other):
+        """Use this rotation to reduce the other.
+
+        The reduction is done as follows:
+        1. For each rotation in `other` the closest rotation in `self` is found. 
+           Denote them as ``p`` and ``q`` respectively.
+        2. The "reduced" rotation is computed as the rotation from the found closest 
+           rotation to the given --- ``r = q.inv() * p``.
+
+        Parameters
+        ----------
+        other : `Rotation` instance
+            Object containing the rotations to be reduced.
+
+        Returns
+        -------
+        reduced : `Rotation` instance
+            Object containing reduced rotations.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> r = R.from_quat([0.5, 0.5, 0.5, 0.5])
+        >>> R.create_group("O").reduce(r).as_quat()
+        array([0., 0., 0., 1.])
+
+        >>> R.create_group("C2").reduce(r).as_quat()
+        array([0.5, 0.5, 0.5, 0.5])
+        """
+        p = other
+        qinv = self.inv()
+        A = p.as_quat().reshape((len(p), 4))
+        B = qinv.as_quat().reshape((len(qinv), 4))
+
+        cosines = np.outer(A[:, 3], B[:, 3])
```

Comment:
A short technical explanation of the algorithm will be useful, it's not obvious.

Reply:
I have added an explanation of what is going on here.